### PR TITLE
Feature/setup opt

### PR DESCRIPTION
**Pull request by xross**
Optimisation for handling SETUP token. This removed separate STALL table and moves checking to a less critical region of the code.

Added a missing test for when an EP is not ready. I was changing code in this area and seemed prudent to add a simple test.

### CODE REVIEW

File: lib_xud/src/core/XUD_Main.xc [hunk 1]
```diff
@@ -487,7 +505,7 @@ int XUD_Main(chanend c_ep_out[], int noEpOut,
 
     for(int i = 0; i < USB_MAX_NUM_EP_IN; i++)
     {
-        handshakeTable_IN[i] = USB_PIDn_NAK;
+        handshakeTable_IN[i] = 0;//USB_PIDn_NAK;
```
Comment (DaveAtkinsonXMOS) [inline lib_xud/src/core/XUD_Main.xc:508]: This comment is rather confusing. I'm guessing it's just not USB_PIDn_NAK anymore, and the comment should be removed?

**Comment by xross** [comment]
Temp moved to draft. One test fail.

File: tests/test_bulk_notready.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Copyright 2016-2021 XMOS LIMITED.
```
Comment (DaveAtkinsonXMOS) [inline tests/test_bulk_notready.py:1]: If I run this test, I get 8 passes and 8 failures. The failures are all at 600 MHz core speed, is this to be expected?

**Review by DaveAtkinsonXMOS — COMMENTED** [review]
Approval to approve if my comments are not important...

File: tests/test_bulk_notready.py (hunk 2 above)
Comment (xross) [reply tests/test_bulk_notready.py:1]: I had only run smoke tests, thanks for that. Will investigate.

File: lib_xud/src/core/XUD_Main.xc (hunk 1 above)
Comment (xross) [reply lib_xud/src/core/XUD_Main.xc]: Fixed

File: tests/test_bulk_notready.py (hunk 2 above)
Comment (xross) [reply tests/test_bulk_notready.py:1]: Its the FS tests that fail. I have xfailed and will raise an issue.

**Review by DaveAtkinsonXMOS — APPROVED** [review]
LGTM

**Pull request merged**